.builds/freebsd.yml

```diff
@@ -15,6 +15,10 @@ environment:
 tasks:
 - setup: |
     cd Nim
+    if [ "${GITHUB_PR_NUMBER+1}" ]; then
+      # this was a PR, so we need to checkout devel and try to merge into it
+      git pull origin devel
```

Comment:
Are you sure that this would merge Nim's devel? According to dispatch.sr.ht [manual](https://man.sr.ht/dispatch.sr.ht/github.md#build-environment):

> dispatch.sr.ht will check out the specific commit you pushed when submitting builds, and will rewrite the upstream clone URL to the fork when building pull requests.

This leaves the impression that the `origin` here will be the fork.

Also what would happen if PRs are targeted at a different branch? Merging `devel` wouldn't be the right thing to do. It also appears to me that sourcehut doesn't expose the target branch of the PR, so implementing this would be very hard.

Reply:
good points
@euantorano can you experiment by printing a few things for debugging?
* git remote -v
* git branch
* whatever else is needed

maybe we can get away without `pull_request.base.ref` and still be reliable

Reply:
```
+ git remote -v
origin	https://github.com/euantorano/Nim.git (fetch)
origin	https://github.com/euantorano/Nim.git (push)
+ git branch
* (HEAD detached at 87cab5a1d)
  devel
```

Environment variables:

```
  GITHUB_BASE_REPO: nim-lang/Nim
  GITHUB_DELIVERY: 045791a0-78df-11eb-9b54-6a6cfb127ffa
  GITHUB_EVENT: pull_request
  GITHUB_HEAD_REPO: euantorano/Nim
  GITHUB_PR_NUMBER: '17104'
  GITHUB_PR_TITLE: 'fix #17097 -builds.sr.ht  CI: merge PRs into devel before testing'
```

Reply:
how about:
```
git remote add upstream https://github.com/nim-lang/Nim
git pull upstream devel
```

Reply:
wait, you can use `GITHUB_BASE_REPO` to infer the upstream url (https://github.com/nim-lang/Nim), so at least you can use something like:

git remote add upstream https://github.com/$GITHUB_BASE_REPO
(not sure about devel, maybe hardcode for now)